src/nvim/search.c

```diff
@@ -4347,121 +4347,288 @@ int linewhite(linenr_T lnum)
+                                int show_top_bot_msg, char_u  *msgbuf,
+                                int recompute, int maxcount, long timeout)
 {
-    int       save_ws = p_ws;
-    int       wraparound = false;
-    pos_T     p = (*pos);
-    static  pos_T   lastpos = { 0, 0, 0 };
+    searchstat_T stat;
+
+    update_search_stat(dirc, pos, cursor_pos, &stat, recompute, maxcount,
+                       timeout);
+    if (stat.cur > 0) {
+      char  t[SEARCH_STAT_BUF_LEN];
+      size_t  len;
+
+      if (curwin->w_p_rl && *curwin->w_p_rlc == 's') {
+        if (stat.incomplete == 1) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[?/??]");
+        } else if (stat.cnt > maxcount && stat.cur > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[>%d/>%d]",
+                       maxcount, maxcount);
+        } else if (stat.cnt > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[>%d/%d]",
+                       maxcount, stat.cur);
+        } else {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[%d/%d]",
+                       stat.cnt, stat.cur);
+        }
+      } else {
+        if (stat.incomplete == 1) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[?/??]");
+        } else if (stat.cnt > maxcount && stat.cur > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[>%d/>%d]",
+                       maxcount, maxcount);
+        } else if (stat.cnt > maxcount) {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[%d/>%d]",
+                       stat.cur, maxcount);
+        } else {
+          vim_snprintf(t, SEARCH_STAT_BUF_LEN, "[%d/%d]",
+                       stat.cur, stat.cnt);
+        }
+      }
+
+      len = STRLEN(t);
+      if (show_top_bot_msg && len + 2 < SEARCH_STAT_BUF_LEN) {
+        memmove(t + 2, t, len);
+        t[0] = 'W';
+        t[1] = ' ';
+        len += 2;
+      }
+
+      memmove(msgbuf + STRLEN(msgbuf) - len, t, len);
+      if (dirc == '?' && stat.cur == maxcount + 1) {
+        stat.cur = -1;
+      }
+
+      // keep the message even after redraw, but don't put in history
+      msg_hist_off = true;
+      msg_ext_set_kind("search_count");
+      give_warning(msgbuf, false);
+      msg_hist_off = false;
+    }
+}
+
+// Add the search count information to "stat".
+// "stat" must not be NULL.
+// When "recompute" is TRUE always recompute the numbers.
+// dirc == 0: don't find the next/previous match (only set the result to "stat")
+// dirc == '/': find the next match
+// dirc == '?': find the previous match
+static void update_search_stat(int dirc, pos_T *pos, pos_T *cursor_pos,
+                               searchstat_T *stat, int recompute, int maxcount,
+                               long timeout)
+{
+    int             save_ws = p_ws;
+    int             wraparound = false;
+    pos_T           p = (*pos);
+    static pos_T    lastpos = { 0, 0, 0 };
     static int      cur = 0;
     static int      cnt = 0;
+    static int      exact_match = false;
```

Comment:
It's documented as a boolean.
`exact_match` uses true/false values only.
Why is this int?

Reply:
because I'm blind 🤦‍♂️